src/Kafka/ConsumerGroups/components/OffsetSelect.tsx

```diff
@@ -0,0 +1,61 @@
+import { Select, SelectOption, SelectVariant } from "@patternfly/react-core";
+import type { SelectProps } from "@patternfly/react-core";
+import { useState } from "react";
+import type { FunctionComponent } from "react";
+import { useTranslation } from "react-i18next";
+import type { OffsetValue } from "../types";
+
+export type OffsetSelectProps = {
+  value: OffsetValue;
+  onChange: (value: OffsetValue) => void;
+};
+
+export const OffsetSelect: FunctionComponent<OffsetSelectProps> = ({
+  onChange,
+  value,
+}) => {
+  const { t } = useTranslation("kafka");
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onToggle = (isOpen: boolean) => {
+    setIsOpen(isOpen);
+  };
+
+  const offsetValueOption: { [key in OffsetValue]: string } = {
+    absolute: t("consumerGroup.offset.absolute"),
+    latest: t("consumerGroup.offset.latest"),
+    earliest: t("consumerGroup.offset.earliest"),
+    timestamp: t("consumerGroup.offset.timestamp"),
+  };
+
+  const onSelect: SelectProps["onSelect"] = (_, selection) => {
+    onChange(selection as OffsetValue);
+    setIsOpen(false);
+  };
+
+  const makeOptions = () => {
+    return Object.entries(offsetValueOption).map(([value, label]) => (
+      <SelectOption key={value} value={value}>
+        {label}
+      </SelectOption>
+    ));
+  };
+
+  return (
+    <Select
+      aira-label={t("consumerGroup.reset_offset_new_offset_label")}
+      toggleAriaLabel={t("consumerGroup.reset_offset_new_offset_label")}
+      id={"topic-select"}
+      onToggle={onToggle}
+      onSelect={onSelect}
+      variant={SelectVariant.single}
+      isOpen={isOpen}
+      placeholderText={t("common:select")}
+      width={200}
```

Comment:
Can you set the width here to be 100% and then set `maxWidth="600px"` on the parent `<Form>`? 

Otherwise, if you set the width here, we would want something like width=100% and max-width=400px. I'm not sure if those are both available as props on the select.

Reply:
@riccardo-forina ```
All form inputs should either be the same length. Sometimes form inputs for values that are expected to be a few characters will be intentionally short, but I'm not sure that really applies to our case with the New offset input. You can see more details about form input width in these PF guidelines: https://www.patternfly.org/v4/components/form/design-guidelines#forms-on-a-page
I would rather have the form input slightly too long rather than too short and end up cutting off topic names that are long
I'm thinking that a max-width of 400px would be good, and a width of 100%. This way, if the modal is smaller than 400px, it displays at 100% instead of getting cut off by the modal.
``` These are the thoughts of @jgiardino on select input length and also she suggested taking your opinion on this, Could you please provide your feedback

Reply:
That makes sense, @riccardo-forina! I also realize that the guidance from PF is specifically for a form on a page: 
> To accommodate lengthy input fields, the maximum suggested width is 880px. Use your best judgement based on the context and content of your form –– 880px is typically too wide for most use cases.

but now that I scroll down to the section about modals, it's recommending what you suggest above: 
> Forms in a modal should be horizontally centered with the modal background and fill the modal width, leaving 24px between the form content and the modal background edges.